src/BenchmarkDotNet.Diagnostics.Windows/NativeMemoryDiagnoser.cs

```diff
@@ -0,0 +1,257 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using BenchmarkDotNet.Analysers;
+using BenchmarkDotNet.Diagnosers;
+using BenchmarkDotNet.Diagnostics.Windows.Tracing;
+using BenchmarkDotNet.Engines;
+using BenchmarkDotNet.Exporters;
+using BenchmarkDotNet.Loggers;
+using BenchmarkDotNet.Reports;
+using BenchmarkDotNet.Running;
+using BenchmarkDotNet.Validators;
+using JetBrains.Annotations;
+using Microsoft.Diagnostics.Symbols;
+using Microsoft.Diagnostics.Tracing.Etlx;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Parsers.Kernel;
+using Address = System.UInt64;
+
+namespace BenchmarkDotNet.Diagnostics.Windows
+{
+    public class NativeMemoryDiagnoser : IDiagnoser
+    {
+        private static readonly string LogSeparator = new string('-', 20);
+        internal readonly LogCapture Logger = new LogCapture();
+        private readonly EtwProfiler etwProfiler;
+
+        [PublicAPI] // parameterless ctor required by DiagnosersLoader to support creating this profiler via console line args
+        public NativeMemoryDiagnoser()
+        {
+            etwProfiler = new EtwProfiler(CreateDefaultConfig());
+        }
+
+        public IEnumerable<string> Ids => new[] { nameof(NativeMemoryDiagnoser) };
+
+        public IEnumerable<IExporter> Exporters => Array.Empty<IExporter>();
+
+        public IEnumerable<IAnalyser> Analysers => Array.Empty<IAnalyser>();
+        public void DisplayResults(ILogger logger)
+        {
+            logger.WriteLineHeader(new string('-', 20));
+            foreach (var line in Logger.CapturedOutput)
+                logger.Write(line.Kind, line.Text);
+        }
+
+        public void Handle(HostSignal signal, DiagnoserActionParameters parameters)
+        {
+            etwProfiler.Handle(signal, parameters);
+        }
+
+        //Code is inspired by https://github.com/Microsoft/perfview/blob/master/src/PerfView/PerfViewData.cs#L5719-L5944
+        private (long totalMemory, long memoryLeak) ParseEtlFile(BenchmarkCase parameters)
```

Comment:
this method is very big, you could consider moving it to a separate file, similar to `TraceLogParser`

Reply:
TODO.